airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/test/commands.py

```diff
@@ -3,167 +3,127 @@
-from pipelines.consts import DOCKER_HOST_NAME, DOCKER_HOST_PORT, DOCKER_VERSION
-from pipelines.dagger.actions.system import docker
-from pipelines.helpers.utils import sh_dash_c
+from pipelines.helpers.git import get_modified_files
+from pipelines.helpers.utils import transform_strs_to_paths
 from pipelines.models.contexts.click_pipeline_context import ClickPipelineContext, pass_pipeline_context
+from pipelines.models.steps import StepStatus
 
 if TYPE_CHECKING:
-    from typing import List, Tuple
+    from pathlib import Path
+    from typing import List, Set, Tuple
 
-    import dagger
 
-## HELPERS
-async def run_poetry_command(container: dagger.Container, command: str) -> Tuple[str, str]:
-    """Run a poetry command in a container and return the stdout and stderr.
+async def find_modified_internal_packages(pipeline_context: ClickPipelineContext) -> Set[Path]:
+    """Finds the modified internal packages according to the modified files on the branch/commit.
 
     Args:
-        container (dagger.Container): The container to run the command in.
-        command (str): The command to run.
+        pipeline_context (ClickPipelineContext): The context object.
 
     Returns:
-        Tuple[str, str]: The stdout and stderr of the command.
+        Set[Path]: The set of modified internal packages.
     """
-    container = container.with_exec(["poetry", "run", *command.split(" ")])
-    return await container.stdout(), await container.stderr()
+    modified_files = transform_strs_to_paths(
+        await get_modified_files(
+            pipeline_context.params["git_branch"],
+            pipeline_context.params["git_revision"],
+            pipeline_context.params["diffed_branch"],
+            pipeline_context.params["is_local"],
+            pipeline_context.params["ci_context"],
+        )
+    )
+    modified_packages = set()
+    for modified_file in modified_files:
+        for internal_package in INTERNAL_POETRY_PACKAGES_PATH:
+            if modified_file.is_relative_to(internal_package):
+                modified_packages.add(internal_package)
+    return modified_packages
+
+
+async def get_packages_to_run(pipeline_context: ClickPipelineContext) -> Set[Path]:
+    """Gets the packages to run the poe tasks on.
+
+    Args:
+        pipeline_context (ClickPipelineContext): The context object.
+
+    Raises:
+        click.ClickException: If no packages are specified to run the poe tasks on.
+
+    Returns:
+        Set[Path]: The set of packages to run the poe tasks on.
+    """
+    if not pipeline_context.params["poetry_package_paths"] and not pipeline_context.params["modified"]:
+        raise click.ClickException("You must specify at least one package to test.")
+
+    poetry_package_paths = set()
+    if pipeline_context.params["modified"]:
+        poetry_package_paths = await find_modified_internal_packages(pipeline_context)
+
+    return poetry_package_paths.union(set(pipeline_context.params["poetry_package_paths"]))
```

Comment:
[the command description says --modified should be a filter to only run on modified internal packages](https://github.com/airbytehq/airbyte/pull/34736/files#diff-a27b9e26d482f1d0ed938f980451495f4fd86a9ff8f88d4c654676f60279ad37R92), but this seems to run both on the explicitly specific packages as well as the modified ones.

Which behavior is correct?